src/test/scala/com/dwolla/sqlserver/init/ArbitraryRefinedTypes.scala

```diff
@@ -0,0 +1,59 @@
+package com.dwolla.sqlserver.init
+
+import cats._
+import cats.syntax.all._
+import com.dwolla.sqlserver.init.repositories.ReservedWords
+import eu.timepit.refined.api.{Refined, Validate}
+import eu.timepit.refined.refineV
+import org.scalacheck.{Arbitrary, Gen, Shrink}
+
+trait ArbitraryRefinedTypes {
+  def refinedConst[A] = new PartiallyAppliedRefinedConst[A]
+
+  implicit val shrinkSqlIdentifier: Shrink[SqlIdentifier] = Shrink.shrinkAny
+  def genSqlIdentifier[F[_] : Applicative]: Gen[F[SqlIdentifier]] =
+    for {
+      initial <- Gen.alphaChar
+      len <- Gen.chooseNum(0, 27) // see comment on SqlIdentifier
+      tail <- Gen.stringOfN(len, Gen.oneOf(Gen.alphaChar, Gen.numChar, Gen.const('_')))
+      identifier = s"$initial$tail"
+      if !ReservedWords.contains(identifier)
+      refined <- refineV[SqlIdentifierPredicate](s"$initial$tail").fold(_ => Gen.fail, Gen.const)
+    } yield refined.pure[F]
+  implicit val arbSqlIdentifier: Arbitrary[SqlIdentifier] = Arbitrary(genSqlIdentifier[Id])
+
+  implicit val shrinkSqlServerUser: Shrink[SqlServerUser] = Shrink.shrinkAny
+  def genSqlServerUser[F[_] : Applicative]: Gen[F[SqlServerUser]] =
+    for {
+      initial <- Gen.alphaChar
+      len <- Gen.chooseNum(0, 63)
+      tail <- Gen.stringOfN(len, Gen.oneOf(Gen.alphaChar, Gen.numChar, Gen.const('_')))
+      user = s"$initial$tail"
+      if !ReservedWords.contains(user)
+      refined <- refineV[SqlServerUserPredicate](s"$initial$tail").fold(_ => Gen.fail, Gen.const)
+    } yield refined.pure[F]
+  implicit val arbSqlServerUser: Arbitrary[SqlServerUser] = Arbitrary(genSqlServerUser[Id])
+
+  implicit val shrinkGeneratedPassword: Shrink[GeneratedPassword] = Shrink.shrinkAny
+  def genGeneratedPassword[F[_] : Applicative]: Gen[F[GeneratedPassword]] = {
+    val allowedPunctuation: Gen[Char] = Gen.oneOf("""! " # $ % & ( ) * + , - . / : < = > ? @ [ ] ^ _ { | } ~ """.replaceAll(" ", "").toList)
+    val allowedCharacters: Gen[Char] = Gen.oneOf(Gen.alphaChar, Gen.numChar, allowedPunctuation)
+
+    for {
+      initial <- allowedCharacters
```

Comment:
Instead of these first four lines, I think we could do
```scala
      length <- Gen.chooseNum(8, 128)
      password <- Gen.stringOfN(length, allowedCharacters)
```
to simplify this logic and ensure that we always generate at least a minimum-length password.

Reply:
Kind of addressed this.... one of the tests failed when a password didn't contain any letters so we added some `suchThat()`s